File: selector/src/main/java/com/luck/picture/lib/widget/MediumBoldTextView.java
package com.luck.picture.lib.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.text.TextPaint;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

import com.luck.picture.lib.R;

/**
 * @author：luck
 * @date：2020/8/25 10:32 AM
 * @describe：MediumBoldTextView
 */
public class MediumBoldTextView extends AppCompatTextView {
    private float mStrokeWidth = 0.6F;

    public MediumBoldTextView(Context context) {
        this(context, null);
    }

    public MediumBoldTextView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public MediumBoldTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable.PictureMediumBoldTextView, defStyleAttr, 0);
        mStrokeWidth = a.getFloat(R.styleable.PictureMediumBoldTextView_stroke_Width, mStrokeWidth);
        a.recycle();
    }

    @Override
    protected void onDraw(Canvas canvas) {
        TextPaint paint = getPaint();
        if (paint.getStrokeWidth() != mStrokeWidth) {
            paint.setStrokeWidth(mStrokeWidth);
            paint.setStyle(Paint.Style.FILL_AND_STROKE);
        }
        super.onDraw(canvas);
    }

    public void setStrokeWidth(float mStrokeWidth) {
        this.mStrokeWidth = mStrokeWidth;
        invalidate();
    }
}

File: selector/src/main/java/com/luck/picture/lib/widget/MediumBoldTextView.kt
package com.luck.picture.lib.widget

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.luck.picture.lib.R

/**
 * @author：luck
 * @date：2020/8/25 10:32 AM
 * @describe：MediumBoldTextView
 */
open class MediumBoldTextView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatTextView(context, attrs, defStyleAttr) {
    private var mStrokeWidth = 0.6f
    override fun onDraw(canvas: Canvas) {
        val paint = paint
        if (paint.strokeWidth != mStrokeWidth) {
            paint.strokeWidth = mStrokeWidth
            paint.style = Paint.Style.FILL_AND_STROKE
        }
        super.onDraw(canvas)
    }

    fun setStrokeWidth(mStrokeWidth: Float) {
        this.mStrokeWidth = mStrokeWidth
        invalidate()
    }

    init {
        val a = context.theme.obtainStyledAttributes(
            attrs,
            R.styleable.SelectorMediumBoldTextView,
            defStyleAttr,
            0
        )
        mStrokeWidth = a.getFloat(R.styleable.SelectorMediumBoldTextView_stroke_Width, mStrokeWidth)
        a.recycle()
    }
}